File: with_node/src/day3.test.ts
import assert from "node:assert";
import test from "node:test";

import { readInput } from "./readInput.ts";
import { part1, part2 } from "./day3.ts";

test("day3...", async (t) => {
  await t.test("part1 says...", () => {
    assert.strictEqual(part1(readInput("day3part1sample")), 161);
  });

  await t.test("part2 says...", () => {
    assert.strictEqual(part2(readInput("day3part2sample")), 48);
  });
});

File: 2024/with_elixir/test/day3_test.exs
defmodule Day3Test do
  use ExUnit.Case
  doctest Day3

  import ReadInput

  test "part1..." do
    assert Day3.part1(read_input("day3part1sample")) == 161
  end

  test "part2..." do
    assert Day3.part2(read_input("day3part2sample")) == 48
  end
end
